gluon-mesh-vpn-wireguard-vxlan/files/lib/gluon/gluon-mesh-wireguard/checkuplink

```diff
@@ -0,0 +1,88 @@
+#!/bin/sh
+
+xor2() {
+        echo -n "${1:0:1}"
+        echo -n "${1:1:1}" | tr '0123456789abcdef' '23016745ab89efcd'
+}
+
+interface_linklocal() {
+        # We generate a predictable v6 address
+        local macaddr="$(echo $(uci get wireguard.wireguard.privatekey | wg pubkey) |md5sum|sed 's/^\(..\)\(..\)\(..\)\(..\)\(..\).*$/02:\1:\2:\3:\4:\5/')"
+        local oldIFS="$IFS"; IFS=':'; set -- $macaddr; IFS="$oldIFS"
+        echo "fe80::$(xor2 "$1")$2:$3ff:fe$4:$5$6"
+}
+
+#Wurde bereits ein privatekey generiert?
+temp=$(uci get wireguard.wireguard.privatekey);
+#Falls nicht, wird das hier nachgeholt
+if [ "$?" -ne "0" ]; then
+        uci set wireguard.wireguard.privatekey=$(wg genkey);
+        uci commit wireguard
+fi
+
+#Ist wireguard überhaupt aktiviert??
+if [ "$(uci get wireguard.wireguard.enabled)" == "true" ] || [ "$(uci get wireguard.wireguard.enabled)" == "1" ]; then
+
+        #Ab hier kommen die Tests, ob die Verbindung steht
+        CONNECTED=1
+
+        INTERFACE=$(uci get wireguard.wireguard.iface)
+
+        # Wireguardinterface vorhanden???
+        ip addr show dev $INTERFACE > /dev/null
+        if [ $? != 0 ]; then
+                CONNECTED=0
+        fi
+
+        # VXLAN Interface vorhanden??
+        if [ $CONNECTED != 0 ]; then
+                ip addr show dev vx_mesh_vpn > /dev/null
+                if [ $? != 0 ]; then
+                        CONNECTED=0
+                fi
+        fi
+
+        # Funktioniert das VXLAN Interface ueberhaupt? Also gehen Daten rueber?
+        if [ $CONNECTED != 0 ]; then
+                RXBYTES=$(ip -statistics link show dev vx_mesh_vpn |  sed '4q;d' | awk '{print $1}')
+                if [ $RXBYTES == 0 ]; then
+                        CONNECTED=0
+                fi
+        fi
+
+        # Und funktioniert das Wireguardinterface? Kann ich den Server erreichen?
+        if [ $CONNECTED != 0 ]; then
+                ping -c1 -w2  $(wg  | grep fe80 | cut -d'/' -f1 | awk '{print $3}')%$INTERFACE
+                if [ $? != 0 ]; then
+                        CONNECTED=0
+                fi
+        fi
+	
+        # Wenn die Tests fehlgeschlagen sind, neu connecten
+        if [ $CONNECTED == 0 ]; then                                    
+
+                NUMBER_OF_PEERS=$(uci show wireguard | egrep -ce peer_[0-9].endpoint)
```

Comment:
Should use quite mode `uci -q`

Can peers be disabled? If so, this would be a potential issue.